tests/schema/test.permissions.js

```diff
@@ -24,30 +24,31 @@ describe('/permissions', () => {
   });
 
   it('should not allow an invalid permission', () => {
-    var manifest = cloneDeep(validManifest);
+    const manifest = cloneDeep(validManifest);
     manifest.permissions = ['wat'];
     validate(manifest);
     assertHasMatchingError(validate.errors, {
       dataPath: '/permissions/0',
     });
   });
 
-  var matchingPatterns = [
+  const matchingPatterns = [
     '*://developer.mozilla.org/*',
     'http://developer.mozilla.org/*',
-    'https://foo.com',
-    'ftp://do.people.still.use.this',
-    'app://wat',
+    // FIXME: the loop in this test was previously broken.
```

Comment:
This is the bad loop that needs a bug filing to fix. The `matchPattern` seen in the test was always the last value in the list.